function/builtin/transform/special.go

```diff
@@ -88,11 +87,13 @@ var ExponentialMovingAverage = function.MakeFunction(
 	"transform.exponential_moving_average",
 	func(context function.EvaluationContext, listExpression function.Expression, size time.Duration) (api.SeriesList, error) {
 		// Applying a similar trick as did TimeshiftFunction. It fetches data prior to the start of the timerange.
-		limit := int(float64(size)/float64(context.Timerange.Resolution()) + 0.5) // Limit is the number of (additional) items to include in the average
-
-		timerange := context.Timerange
+		limit := int(float64(size)/float64(context.Timerange().Resolution()) + 0.5) // Limit is the number of items to include in the average
+		if limit < 1 {
```

Comment:
There wasn't: I just added some, and also made the moving average functions error if they're given a negative duration (0 is fine and leaves the input unchanged).